azplugins/flow.py

```diff
@@ -673,31 +674,35 @@ class FlowProfiler:
         hoomd.analyze.callback(f, period=1e3)
         hoomd.run(1e4)
         if hoomd.comm.get_rank() == 0:
-            np.savetxt('profiles.dat', np.column_stack((f.centers, f.density, f.velocity)))
+            np.savetxt('profiles.dat', np.column_stack((f.centers, f.number_density, f.number_velocity[:,2], f.kT)))
 
     .. note::
 
         In MPI simulations, the profiles are **only** valid on the root rank. Accessing them on
         other ranks will produce an error.
 
+    .. note::
+
+        The temperature profile `kT` is calculated from the velocity values without substracting any shear flow.
```

Comment:
We discussed it briefly -- I remember a vaguely that we wanted to keep the flow profiler generic and applicable to different types of flow. But, we should be able to do that anyway by subtracting the average velocity in each bin and looking at the velocity variance, correct? Is information about the flow (direction, symmetry etc) needed for calculating kT?

Reply:
Yeah, I don’t remember, but the profile unbiased temperature makes the most sense to me. You don’t need any extra information about the flow, I think. You compute the kinetic energy directly from the velocities, then subtract the kinetic energy of the center of mass of the bin, like here:

https://github.com/glotzerlab/hoomd-blue/blob/ed60448e7720fe6bade26ca53876259d93eec9ea/hoomd/mpcd/CellThermoCompute.cc#L340

You already have all the information needed for the center-of-mass KE since you know the total mass and the mass-average velocity of the bin.

The temperature is zero if there are 0 or 1 particles in the cell.